redis/redis.go

```diff
@@ -147,22 +145,33 @@ func (c *Client) PublishEnabled(state *EnabledMessage) {
 		})
 
 		if mErr == nil {
-			c.client.Publish(c.ctx, SyncChannelName, binMsg)
+			c.client.Publish(ctx, SyncChannelName, binMsg)
 		}
 	}
 }
 
 // GetRedisCache reads the redis cache and publish it to the channel
-func (c *Client) GetRedisCache() {
+func (c *Client) GetRedisCache(ctx context.Context) {
 	c.l.Debug("GetRedisCache")
 
 	go func() {
-		iter := c.client.Scan(c.ctx, 0, prefixKey("*"), 0).Iterator()
-		for iter.Next(c.ctx) {
-			response, err := c.getResponse(iter.Val())
+		iter := c.client.Scan(ctx, 0, prefixKey("*"), 0).Iterator()
+		if err := iter.Err(); err != nil {
+			c.l.Error("GetRedisCache ", err)
+
+			return
+		}
+
+		for iter.Next(ctx) {
+			response, err := c.getResponse(ctx, iter.Val())
 			if err == nil {
 				if response != nil {
-					c.CacheChannel <- response
+					select {
+					case <-ctx.Done():
+						return
+					default:
+						c.CacheChannel <- response
+					}
```

Comment:
I think we're better off checking the context first thing in the loop body: here if `getResponse` fails because the context is done, we'll never exit the loop.  
In practice the program will still exit could be an issue if we someday want to recreate the client while blocky is running.

Also to check a context in fewer lines you can do:
```go
if err := ctx.Err(); err != nil {
	// ...
}
```

Reply:
I disagree on this one 😅

If getResponse fails the next call would be Next which is also context aware and would terminate the loop.
And if not the redis client would be closed in the startup go routine which would force Next to terminate the loop.

In my opinion the check is at the right place since all other function calls are already context aware.

Reply:
> Also to check a context in fewer lines you can do:
> ```go
> if err := ctx.Err(); err != nil {
> 	// ...
> }
> ```

I used that approach first but context checks get easily mixed up with other conditions which made rereading the code troublesome.

I like checking it in a select more since it is more obvious when skimming over code.

But that's only my personal preference I guess. 😅

Reply:
I personally like the shorter one cause it's not the main logic of the function and it makes it less obtrusive, also forces you to think about the error, but I use the channel when there's another channel in play.  
Both are fine anyways, was more to get the coverage complaints down at the same time!